distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerCliTests.java

```diff
@@ -315,6 +320,15 @@ public void testServerExitsNonZero() throws Exception {
         assertThat(exitCode, equalTo(140));
     }
 
+    public void testServerCliCanLoadDifferentSecureSettingsImplementation() throws Exception {
```

Comment:
nit on the naming: everything here is about the ServerCli, so no need for that prefix. I personally find these long names more difficult to read because they are so long. Maybe something as simple as `testSecureSettingsLoaders()`?